### PR TITLE
Ajpal/fix attempts per completed

### PR DESCRIPTION
**Pull request by ajpal**
@dsjen @thallada

### CODE REVIEW

File: analytics_data_api/constants/engagement_events.py [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+    'discussion_contributed',
+    'video_viewed'
+]
+
+EVENTS = [
+    'problem_attempts_per_completed',
+    'problems_attempted',
+    'problems_completed',
+    'discussion_contributions',
+    'videos_viewed'
```
Comment (thallada) [inline analytics_data_api/constants/engagement_events.py]: I get why these change to plural, but I think it would be useful to have a single enum of event types that both database tables could use when referring to the same type of event. This isn't really related to this PR since it existed before, I just wanted to point it out.

**Comment by thallada** [comment]
:+1:

File: analytics_data_api/constants/engagement_types.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
-from analytics_data_api.constants.engagement_entity_types import DISCUSSION, PROBLEM, VIDEO
-from analytics_data_api.constants.engagement_events import ATTEMPTED, COMPLETED, CONTRIBUTED, VIEWED
+from analytics_data_api.constants.engagement_events import (ATTEMPTED, ATTEMPTS_PER_COMPLETED, COMPLETED,
+                                                            CONTRIBUTED, VIEWED, DISCUSSION, PROBLEM, VIDEO)
```
Comment (dsjen) [inline analytics_data_api/constants/engagement_types.py]: Nit: our convention is to import in alphabetical order. :)

File: analytics_data_api/v0/serializers.py [hunk 3]
```diff
@@ -446,23 +445,16 @@ def get_engagement_ranges(self, obj):
...
-                normal_range_queryset = query_set.filter(metric=metric, range_type='normal')
-                high_range_queryset = query_set.filter(metric=metric, range_type='high')
-                engagement_ranges.update({
-                    metric: EnagementRangeMetricSerializer({
-                        'low_range': low_range_queryset[0] if len(low_range_queryset) else None,
-                        'normal_range': normal_range_queryset[0] if len(normal_range_queryset) else None,
-                        'high_range': high_range_queryset[0] if len(high_range_queryset) else None,
-                    }).data
-                })
+        for metric in engagement_events.EVENTS:
```
Comment (dsjen) [inline analytics_data_api/v0/serializers.py]: Much clearer!

File: analytics_data_api/v0/tests/views/test_engagement_timelines.py [hunk 4]
```diff
@@ -9,8 +9,8 @@
 from rest_framework import status
 
 from analyticsdataserver.tests import TestCaseWithAuthentication
-from analytics_data_api.constants.engagement_entity_types import DISCUSSION, PROBLEM, VIDEO
-from analytics_data_api.constants.engagement_events import ATTEMPTED, COMPLETED, CONTRIBUTED, VIEWED
+from analytics_data_api.constants.engagement_events import (ATTEMPTED, COMPLETED, CONTRIBUTED, VIEWED,
```
Comment (dsjen) [inline analytics_data_api/v0/tests/views/test_engagement_timelines.py]: Nit: same alphabetical import convention

File: analytics_data_api/v0/tests/views/test_learners.py [hunk 5]
```diff
@@ -649,12 +640,8 @@ def test_engagement_ranges_only(self):
...
-        actual_events = []
-        for entity_type in actual_entity_types:
-            for event in engagement_events.EVENTS[entity_type]:
-                actual_events.append(event)
-        expected_events = ['contributed', 'attempted', 'attempts_per_completed', 'completed', 'viewed']
-        self.assertEqual(actual_events, expected_events)
+        expected_events = engagement_events.EVENTS
+        response = json.loads(self._get(self.course_id).content)
+        self.assertTrue('engagement_ranges' in response)
+        for event in expected_events:
```
Comment (dsjen) [inline analytics_data_api/v0/tests/views/test_learners.py]: Ah!  It's so much more understandable.

**Comment by dsjen** [comment]
I tested out the data generation and endpoint and it works perfectly.  I left a few "nits."

Thanks, @ajpal!  👍
